SIGSEV with lego migrate

### Welcome

- [x] Yes, I'm using a binary release or the library within the [two latest releases](https://github.com/go-acme/lego/releases).
- [x] Yes, I've searched for similar issues on GitHub and didn't find any.

### How do you use lego?

Docker image

### Effective version of lego

v5.0.1

### Logs

<details>

```console
$ sudo ./lego migrate --path /etc/lego
2026-05-12T21:25:00.341443142+02:00 WARN  The migration will not work if the certificates have been generated with the '--filename' flag. Use the flag '--account-only' to only migrate accounts.
2026-05-12T21:25:00.341624278+02:00 WARN  Please create a backup of "/etc/lego/" before the migration.
Continue? Y/n
Y
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x22c7d58]

goroutine 1 [running]:
github.com/go-acme/lego/v5/cmd/internal/migrate.Accounts({0x7fff2bb916e4?, 0x2ccef07?}, 0x21ffd88ce180)
	github.com/go-acme/lego/v5/cmd/internal/migrate/accounts.go:67 +0x2f8
github.com/go-acme/lego/v5/cmd.createMigrate.func1({0x21ffd88de588?, 0x2e478b0?}, 0x21ffd88de588)
	github.com/go-acme/lego/v5/cmd/cmd_migrate.go:49 +0x236
github.com/urfave/cli/v3.(*Command).run(0x21ffd88de588, {0x2e478b0, 0x21ffd88ad860}, {0x21ffd88ad6e0, 0x3, 0x3})
	github.com/urfave/cli/v3@v3.8.0/command_run.go:364 +0x2f19
github.com/urfave/cli/v3.(*Command).run(0x21ffd88de848, {0x2e478b0, 0x21ffd88ac4e0}, {0x21ffd8904080, 0x4, 0x4})
	github.com/urfave/cli/v3@v3.8.0/command_run.go:298 +0x2745
github.com/urfave/cli/v3.(*Command).Run(...)
	github.com/urfave/cli/v3@v3.8.0/command_run.go:93
main.main()
	github.com/go-acme/lego/v5/main.go:23 +0xa5
```

</details>

### What did you expect to see?

I was expecting the migrate command to succeed as per https://go-acme.github.io/lego/migration/cli/index.html.

### What did you see instead?

I got a SIGSEV.

### Reproduction steps

I'm using Lego through Docker container on an amd64 VyOS machine (https://docs.vyos.io/en/latest/configuration/container/). I was using v4 and updated to v5. When trying to run the lego migrate command, I got a SIGSEV.

I also tried to run the binary directly, and got the exact same result.

I worked around the issue by manually fixing the directory layout as per the migration guide. In doing so, I noticed the presence of a faulty .key file in the new (v5) location, probably due to a previous attempt to renew the certificate prior to migration. 

### Go environment (if applicable)

<details>

```console
# paste output here
```

</details>

### In which context are you using lego?

Personal use

### 🌱 Supporting the Maintainer

- [x] Yes, I find this project useful, and it has saved me time in my workflow/business.
- [ ] Yes, I have [starred](https://github.com/go-acme/lego) this repository to help others find it.
- [ ] Yes, I have [sponsored](https://donate.ldez.dev) this project to keep the project stable and maintained.

### Confirmation

- [x] Yes, I've included all the information above (version, usage, etc.).

--- a/cmd/internal/migrate/accounts.go
+++ b/cmd/internal/migrate/accounts.go
@@ -63,10 +63,15 @@ func Accounts(root string, cfg *configuration.Configuration) error {
 			Email:  oldAccount.Email,
 			Server: guessServer(serverDir),
 			Origin: storage.OriginMigration,
-			Registration: &acme.ExtendedAccount{
+		}
+
+		if oldAccount.Registration != nil {
+			account.Registration = &acme.ExtendedAccount{
 				Account:  oldAccount.Registration.Body,
 				Location: oldAccount.Registration.URI,
-			},
+			}
+		} else {
+			log.Warn("No registration was found for the account.", slog.String("account", accountID))
 		}
 
 		srcKeyPath := filepath.Join(accountDir, oldKeysFolderName, account.GetID()+storage.ExtKey)

